addon/-private/syntax-mount-ext.js

```diff
@@ -0,0 +1,124 @@
+import Ember from 'ember';
+import emberRequire from './ext-require';
+
+const {
+  assert,
+  runInDebug
+} = Ember;
+
+const mountSyntaxModule = Ember.__loader.require('ember-glimmer/syntax/mount');
+const ComponentDefinition = emberRequire('@glimmer/runtime', 'ComponentDefinition');
+const UNDEFINED_REFERENCE = emberRequire('@glimmer/reference', 'UNDEFINED_REFERENCE');
+const RootReference = emberRequire('ember-glimmer/utils/references', 'RootReference');
+const generateControllerFactory = emberRequire('ember-routing', 'generateControllerFactory');
+const OutletLayoutCompiler = emberRequire('ember-glimmer/syntax/outlet', 'OutletLayoutCompiler');
+const FACTORY_FOR = emberRequire('container', 'FACTORY_FOR');
+const AbstractManager = emberRequire('ember-glimmer/syntax/abstract-manager');
+
+function dynamicEngineFor(vm, symbolTable) {
+  let env     = vm.env;
+  let args    = vm.getArgs();
+  let nameRef = args.positional.at(0);
+
+  return new DynamicEngineReference({ nameRef, env, symbolTable });
+}
+
+mountSyntaxModule.mountMacro = function(path, params, hash, builder) {
+  assert(
+    'You can only pass a single argument to the {{mount}} helper, e.g. {{mount "chat-engine"}}.',
+    params.length === 1 && hash === null
+  );
+
+  assert(
+    'The first argument of {{mount}} must be quoted, e.g. {{mount "chat-engine"}}.',
+    typeof params[0] === 'string'
+  );
```

Comment:
This PR is simply a refactor of existing functionality. The additional functionality will be added in follow up PRs.